src/wazuh_modules/syscollector/syscollector_windows.c

```diff
@@ -1052,108 +1034,38 @@ void read_win_program(const char * sec_key, int arch, int root_key, int usec, co
                 cJSON_AddStringToObject(package, "architecture", "unknown");
 
             // Get version
-
-            version = (char *)malloc(TOTALBYTES);
-            cbData = buffer_size;
-
-            ret = RegQueryValueEx(program_key, "DisplayVersion", NULL, NULL, (LPBYTE)version, &cbData);
-            while (ret == ERROR_MORE_DATA) {
-
-                // Increase buffer length
-
-                buffer_size += BYTEINCREMENT;
-                os_realloc(version, buffer_size, version);
-                cbData = buffer_size;
-                ret = RegQueryValueEx(program_key, "DisplayVersion", NULL, NULL, (LPBYTE)version, &cbData);
-            }
-
-            if (ret == ERROR_SUCCESS && version[0] != '\0') {
-                cJSON_AddStringToObject(package, "version", version);
-            }
-
-            free(version);
+            version = w_reg_query_value(program_key, L"DisplayVersion");
+            if (version && *version) cJSON_AddStringToObject(package, "version", version);
```

Comment:
Please use the coding convention. It will make the code more readable:
```c
if (version && *version) {
    cJSON_AddStringToObject(package, "version", version);
}
```

Reply:
Fixed in latest commit. Thanks!